network/data_link/at_params.h

```diff
@@ -0,0 +1,265 @@
+	SOCKET_TCP,
+	/** UDP Connection */
+	SOCKET_UDP
+};
+
+/**
+ * @enum cwmod_param
+ * @brief Type of operation for module
+ */
+enum cwmod_param {
+	/** Operate as client */
+	CLIENT = 1,
+	/** Operate as access point */
+	ACCESS_POINT,
+	/** Operate as both */
+	CLIENT_AND_ACCESS_POINT
+};
+
+/**
+ * @struct cwjap_param
+ * @brief Parameter used to connect to a wifi network
+ */
+struct cwjap_param {
+	/** Ssid */
+	struct at_buff	ssid;
+	/** Password */
+	struct at_buff	pwd;
+};
+
+/**
+ * @struct cwsap_param
+ * @brief Parameter used to set an access point
+ */
+struct cwsap_param {
+	/** AP ssid */
+	struct at_buff		ssid;
+	/** AP password  */
+	struct at_buff		pwd;
+	/** AP channel ID */
+	uint32_t		ch_id;
+	/** Encription type */
+	enum encryption_type	encription;
+};
+
+/**
+ * @struct cipstart_param
+ * @brief Parameter used to start a new connection
+ */
+struct cipstart_param {
+	/** Connection id. Used only if \ref MULTIPLE_CONNECTION is set */
+	uint32_t		id;
+	/** Connection type */
+	enum socket_type	soket_type;
+	/** Connection address. Example "mqtt.brocker.com" */
+	struct at_buff		addr;
+	/** Connection port */
+	uint16_t		port;
+	/* Parameters used only for UDP connection */
+	/** Local port. Used only if \ref SOCKET_UDP is used */
+	uint16_t		local_port;
+	/** Udp mode. Used only if \ref SOCKET_UDP is used */
+	enum {
+		UDP_ENTITY_NO_CHANGE,
+		UDP_ENTITY_CHANGE_ONE,
+		UDP_ENTITY_CHANGE_ALLOWED
+	}			udp_mode;
+};
+
+/**
+ * @struct cipsend_param
+ * @brief Structure used to send data over a connection
+ */
+struct cipsend_param {
+	/** Connection id. Used only if \ref MULTIPLE_CONNECTION is set */
+	uint32_t	id;
+	/**
+	 * Data to be sent. data.buff should be the address of the buffer and
+	 * data.len its size.
+	 */
+	struct at_buff	data;
+	/* Parameters used only for UDP connection */
+	/** Remote address.  Used only if \ref SOCKET_UDP is used */
+	struct at_buff	remote_ip;
+	/** Remote port. Used only if \ref SOCKET_UDP is used */
+	uint16_t	remote_port;
+};
+
+/**
+ * @enum connection_type
+ * @brief Parameter used to set the connection type
+ */
+enum connection_type {
+	/** Only one connection available */
+	SINGLE_CONNECTION,
+	/** Multiple connection available */
+	MULTIPLE_CONNECTION
+};
+
+/**
+ * @struct cipserver_param
+ * @brief Parameter used to configure a TCP server
+ */
+struct cipserver_param {
+	/** Command for open or close the server*/
+	enum {
+		CLOSE_SOCKET,
+		OPEN_SOCKET
+	}		action;
+	/** Port */
+	uint16_t	port;
+};
+
+/**
+ * @enum cipmode_param
+ * @brief Not implemented
+ */
+enum cipmode_param {
+	/** Normal mode */
+	NORMAL_MODE,
+	/** Unvernished mode */
+	UNVARNISHED_MODE
+};
+
+/**
+ * @struct cwlap_set_param
+ * @brief Not used. Find an wifi with specific conditions
+ */
+struct cwlap_set_param {
+	/** ssid*/
+	struct at_buff		ssid;
+	/** mac */
+	struct at_buff		mac;
+	/** channel */
+	uint8_t			channel;
+};
+
+#if defined(PARSE_RESULT)
+/* Search networks. Not used in driver*/
```

Comment:
What is the purpose of the things not used? Do they need to be implemented?

Reply:
Nop, I left them for an eventual further developement. Maybe I will need for example to get the status of a connection.

Reply:
Or for scan wifi network if it will be needed